mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -135,13 +135,13 @@ abstract class MLWriter extends BaseReadWrite with Logging {
 }
 
 /**
- * Trait for classes that provide [[MLWriter]].
+ * Trait for classes that provide `MLWriter`.
```

Comment:
@HyukjinKwon Are you sure this is a problem?  I just tried generating the docs using Java 8, and the link with ```[[MLWriter]]``` worked for both Scala and Java.  Maybe it is fine for some cases (like this, where MLWriter is part of the package being compiled) and not for other cases (such as when the class is from a dependency).